util/server/index.ts

```diff
@@ -0,0 +1,149 @@
+  const devfilesWithoutSource = JSON.parse(devfilesUnparsed) as Devfile[];
+  const devfilesWithSource = devfilesWithoutSource.map((devfile: Devfile) => {
+    devfile.sourceRepo = hostName;
+    return devfile;
+  });
+  return devfilesWithSource;
+};
+
+const getRemoteYAML = async (devfileName: string, yamlLocation: string): Promise<string> => {
+  const res = await fetch(`${yamlLocation}/devfiles/${devfileName}`, {
+    headers: { 'Accept-Type': 'text/plain' }
+  });
+  const devfileYAML = (await res.text()) as string;
+  return devfileYAML;
+};
+
+const getLocalYAML = async (devfileName: string, yamlLocation: string): Promise<string> => {
+  const devfileYAMLRelPath = `${yamlLocation}/${devfileName}/devfile.yaml`.split('/');
+  const devfileYAMLAbsPath = path.join(process.cwd(), ...devfileYAMLRelPath);
+  const devfileYAML = (await fs.readFile(devfileYAMLAbsPath, 'utf8')) as string;
+  return devfileYAML;
+};
+
+const getENVHosts = () => {
+  const envHosts = process.env.DEVFILE_REGISTRY_HOSTS?.split('|').filter((host) => host !== '');
+
+  let hosts: Host = {};
+
+  if (envHosts?.length) {
+    envHosts.forEach((envHost) => {
+      const [hostName, sourceType, hostLocation] = envHost.split('>');
+
+      if (!(sourceType === 'url' || sourceType === 'stacks')) {
+        throw Error(
+          'The environment variable DEVFILE_REGISTRY_HOSTS can only accept "url" or "stacks"'
+        );
+      }
+
+      hosts = {
+        ...hosts,
+        [hostName]: {
+          [sourceType]: hostLocation
+        }
+      };
+    });
+  }
+
+  return hosts;
+};
+
+export const getDevfilesJSON = async (): Promise<Devfile[]> => {
+  let hosts: Host = await getConfigFileHosts('/config/devfile-registry-hosts.json');
+  hosts = { ...hosts, ...getENVHosts() };
+
+  if (!Object.keys(hosts).length) {
+    hosts = {
+      Community: {
+        url: 'https://registry.devfile.io'
+      }
+    };
+  }
+
+  let devfiles: Devfile[] = [];
+  await Promise.all(
+    Object.entries(hosts).map(async ([hostName, hostLocation]) => {
+      let extractedDevfiles: Devfile[] = [];
+      if (hostLocation.url) {
+        extractedDevfiles = await getRemoteJSON(hostName, hostLocation.url);
+      }
+      if (hostLocation.stacks) {
+        extractedDevfiles = await getLocalJSON(hostName, hostLocation.stacks);
+      }
+
+      if (extractedDevfiles.length) {
+        devfiles = devfiles.concat(extractedDevfiles);
+      }
+    })
+  );
+
+  return devfiles;
+};
+
+export const getDevfileYAML = async (devfile: Devfile): Promise<getDevfileYAMLReturnType> => {
+  let devfileYAML = null;
+  let devfileJSON = null;
+
+  if (devfile.type !== 'stack') {
+    return { devfileYAML, devfileJSON };
+  }
+
+  let hosts: Host = await getConfigFileHosts('/config/devfile-registry-hosts.json');
+  hosts = { ...hosts, ...getENVHosts() };
+
+  if (!Object.keys(hosts).length) {
+    hosts = {
+      Community: {
+        url: 'https://registry.devfile.io'
```

Comment:
If the tests will end up hitting the URL, we should use staging (https://registry.stage.devfile.io) instead, as we don't want any CI hitting prod

Reply:
fixed